### ISSUE TITLE
Add support for #cached sections in Qute templates

### ISSUE THREAD

**Issue by fbricon** [redhat-developer/quarkus-ls#919]
According to https://quarkus.io/guides/qute-reference#cached-section, if `io.quarkus:quarkus-cache` is in the classpath, qute templates should automatically be able to use a `{#cached}` section, with an optional `key` attribute.

Currently, the `#cached` section is being shown with an error
<img width="478" alt="Screenshot 2023-07-31 at 11 31 56" src="https://github.com/redhat-developer/quarkus-ls/assets/148698/0b0020fb-4201-4091-b65d-8b5eac0ae393">

@mkouba is this another instance of synthetic bean usage?

**Comment by mkouba** [comment]
> @mkouba is this another instance of synthetic bean usage?

No, `{#cached}` is a section registered here: https://github.com/quarkusio/quarkus/blob/main/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/CacheProcessor.java#L13-L21.

In general, any extension or even application code can register a custom section. See also https://quarkus.io/guides/qute-reference#engine-customization.
